### ISSUE TITLE
[Notification] Focus outline now permanent on low contrast notifications

### ISSUE THREAD

**Issue by richardpilot** [carbon-design-system/carbon#6053]
## Environment

Any

## Detailed description

Issue #5737 was raised to changed the focus outline from white to blue, however the 'fix' was incorrect and is now permanently visible rather than only on `:focus` events.

This makes the low contrast button quite distracting and it now has visually different behaviour from high contrast. 

Much like the original defect, it continues to break accessibility as the button still has no focus indicator.

> What WCAG 2.1 checkpoint does the issue violate?
https://www.ibm.com/able/guidelines/ci162/focus_visible.html

## Steps to reproduce the issue

1. Navigate to https://react.carbondesignsystem.com/?path=/story/notifications--inline
2. Check the low contrast knob
3. focus on the close button, there is no visual change

**Comment by tw15egan** [comment]
Will be fixed in https://github.com/carbon-design-system/carbon/issues/6053

**Comment by joshblack** [comment]
Gonna close this out as a duplicate of https://github.com/carbon-design-system/carbon/issues/5977

Should be addressed in #5978 as TJ noted 👀
